service/dap/server.go

```diff
@@ -376,7 +422,20 @@ func (s *Server) serveDAPCodec() {
 			}
 			return
 		}
-		s.handleRequest(request)
+
+		jsonmsg, _ := json.Marshal(message)
+		s.log.Debug("[<- from client]", string(jsonmsg))
+
+		switch m := message.(type) {
+		case dap.RequestMessage:
+			requests <- m
```

Comment:
Is this a potential deadlock? If the request handling loop is waiting for a specific response but the server sends too many requests before the response.

Reply:
True. Added comments to explain the case and potential solutions. 🤞

Reply:
I was thinking more along the lines of doing the send inside a select with a default statement and logging the error if the send can't be done.

Reply:
The infinitely growing queue isn't a solution because infinite doesn't exist and by logging dropped messages you'll at least know that you have this problem.

Reply:
Done.